tests/lib/devicetree/api/src/main.c

```diff
@@ -3503,6 +3514,13 @@ ZTEST(devicetree_api, test_string_idx_token)
 	zassert_equal(STRING_TOKEN_BY_IDX_VAR(DT_NODELABEL(test_str_array_token_1))[2],
 			token_second_idx_two, "");
 
+	/* Index is in range instance #0 corresponds to test_str_array_token_0 */
```

Comment:
No, instance 0 does not have to be test_str_array_token_0. You are relying on whatever happened to work in your build environment today. This is a common error -- please read the doxygen documentation for DT_INST() carefully and rewrite this without making any assumptions not guaranteed by the API.